doc/source/upgrading_guide_upstream.rst

```diff
@@ -0,0 +1,408 @@
+
+In-place upgrades keep the existing data on the server. Generally speaking,
+in-place upgrades involve stopping the server, removing the installed version of
+|MySQL|, installing the new server, and starting it with the same data files.
+
+While in-place upgrades may not be suitable for high-complexity environments,
+they may be appropriate for many simpler scenarios.
+
+It is strongly recommended that you read the following guides to be aware of the
+incompatible changes that could cause the in-place upgrade to fail:
+
+* `Upgrading MySQL <http://dev.mysql.com/doc/refman/5.7/en/upgrading.html>`_
+* `Upgrading from MySQL 5.6 to 5.7 <http://dev.mysql.com/doc/refman/5.7/en/upgrading-from-previous-series.html>`_
+
+Use one of the following approaches to make the in-place upgrade to |Percona
+Server| 5.7 from |MySQL|.
+
+.. contents::
+   :local:
+   :depth: 1
+
+.. _percona-server.in-place-upgrade.repository.upstream:
+
+In-place upgrade using |Percona| repositories
+================================================================================
+
+The recommended way of installing |Percona Server| is by using the |Percona|
+repositories.
+
+Instructions for your system about how to enable |Percona| repositories can be
+found in the section :ref:`installing_from_binaries`.
+
+The in-place upgrade using |Percona| repositories requires the following steps:
+
+.. contents::
+   :local:
+   :depth: 1
+
+.. _percona-server.in-place-upgrade.preliminary.upstream:
+
+Preliminary steps common to all distributions
+--------------------------------------------------------------------------------
+
+.. include:: .res/text/system-service-manager.support.txt
+
+|tip.run-all.root|.
+
+#. Do the full backup (or dump, if possible) a then stop the server. 
+
+   .. code-block:: bash
+
+      $ service mysql stop
+
+The following steps are different for DEB based and RPM based
+distributions. Depending on your operating system, follow the instructions in
+either :ref:`percona-server.mysql.in-place-upgrade.upstream.deb` or in
+:ref:`percona-server.in-place-upgrade.upstream.rpm`.  Then, :ref:`run mysql_upgrade
+<percona-server.mysql-upgrade.run-manually.upstream>`.
+
+.. _percona-server.in-place-upgrade.upstream.deb:
+
+Upgrading from |MySQL| 5.7 for DEB based Linux distributions
+--------------------------------------------------------------------------------
+
+The following steps are relevant to upgrading |MySQL| 5.7 installed on |debian|
+or |ubuntu|.
+
+#. Check which |MySQL| packages are installed:
+
+   .. code-block:: bash
+
+      $ dpkg-query -l 'mysql-*'
+
+#. Remove |MySQL| server leaving data files intact. In the following
+   example, **MYSQL-PACKAGE-NAME** is a placeholder which represents
+   an installed |MySQL| package that you need to remove. For example,
+   if you have |MySQL| Server Community Edition you need to uninstall
+   `mysql-server` at least.
+
+   .. code-block:: bash
+
+      $ apt remove MYSQL-PACKAGE-NAME
+
```

Comment:
it would be more user-friendly if we would supply the line with the package names (instead of a placeholder) that can be copy-pasted to CLI